ext_mongodb.php

```diff
@@ -354,23 +357,34 @@ public function __construct(mixed $filter, array $options = array())
 			$this->query['flags'] = array_key_exists('flags', $options ) ? (int) $options['flags'] : 0;
 			$this->query['limit'] = array_key_exists('limit', $options ) ? (int) $options['limit'] : 0;
 			$this->query['skip'] = array_key_exists('skip', $options ) ? (int) $options['skip'] : 0;
-		}
 
-		if (array_key_exists('modifiers', $options)) {
-			Utils::mustBeArrayOrObject('modifiers', $options['modifiers']);
-			foreach ($options['modifiers'] as $key => $value) {
-				$this->query['query'][$key] = $value;
+			if (array_key_exists('readConcern', $options)) {
+				if (!($options['readConcern'] instanceof \MongoDB\Driver\readConcern)) {
+					throw new \MongoDB\Driver\Exception\InvalidArgumentException(
+						'Expected "readConcern" option to be MongoDB\Driver\ReadConcern, ' .
+						gettype($options['readConcern']) . ' given'
+					);
+				} else {
+					$this->query['readConcern'] = $options['readConcern']->getLevel();
+				}
 			}
-		}
 
-		if (array_key_exists('projection', $options)) {
-			Utils::mustBeArrayOrObject('projection', $options['projection']);
-			$this->query['fields'] = (array) $options['projection'];
-		}
+			if (array_key_exists('modifiers', $options)) {
+				Utils::mustBeArrayOrObject('modifiers', $options['modifiers']);
+				foreach ($options['modifiers'] as $key => $value) {
+					$this->query['query'][$key] = $value;
+				}
+			}
+
+			if (array_key_exists('projection', $options)) {
+				Utils::mustBeArrayOrObject('projection', $options['projection']);
+				$this->query['fields'] = (array) $options['projection'];
```

Comment:
I realize this line wasn't actually changed in the PR, but is there a reason you're using array-casting here instead of object? I assume you'd want the behavior as `sort` to ensure BSON object conversion in the event a user was projecting `{ "0": 1 }`.